src/MiniCover/Commands/Options/MiniCoverOption.cs

```diff
@@ -5,36 +5,33 @@ namespace MiniCover.Commands.Options
 {
     public abstract class MiniCoverOption<TValue> : IMiniCoverOption<TValue>
     {
-        private CommandOption option;
-        private bool validated;
-        private TValue value;
+        protected CommandOption Option;
+        protected bool Validated { get; private set; }
+        protected TValue ValueField { get; private set; }
 
-        protected abstract string Description { get; }
-        protected abstract string OptionTemplate { get; }
-        protected virtual CommandOptionType Type => CommandOptionType.SingleValue;
-
-        public TValue Value
+        public TValue GetValue()
         {
-            get
-            {
-                if (!validated)
-                    throw new MemberAccessException("Option should be validated before Value access");
+            if (!Validated)
+                throw new MemberAccessException("Option should be validated before GetValue access");
 
-                return value;
-            }
+            return ValueField;
         }
 
+        protected abstract string Description { get; }
```

Comment:
Same as previously

```csharp
protected MiniCoverOption(string template, string description, CommandOptionType type = CommandOptionType.SingleValue)
```

Reply:
That is simple :)
Because ... abstract enforce implementation. if it will be a ctor option it will be possible to be mistaken. 

The best it will be a constant but the base class cannot enforce creating constant field/variable

Reply:
big difference is that I can break liskov with your code:
protected override string OptionTemplate => throw new System.Exception();

So I agree that you can switch template vs description but that a problem of primitive obsession we have in c#

Reply:
... nope in this way you can break it in any inheritance. Unfortunately, I disagree at that point.

Reply:
sure you can break also in constructor but in that case you didn't even construct the instance.
The point is more that i have only one inheritance point of failure you give 3.

Reply:
updated